app/assets/javascripts/analytics.js

```diff
@@ -1,55 +1,15 @@
-// Inspired by and modified from
-// http://railsapps.github.io/rails-google-analytics.html
-
-var GoogleAnalytics = (function() {
-  function GoogleAnalytics() {}
-
-  GoogleAnalytics.load = function() {
-    var firstScript, ga;
-    window._gaq = [];
-    GoogleAnalytics.analyticsId = GoogleAnalytics.getAnalyticsId();
-    window._gaq.push(["_setAccount", GoogleAnalytics.analyticsId]);
-    window._gaq.push(['_gat._anonymizeIp']);
-    ga = document.createElement("script");
-    ga.type = "text/javascript";
-    ga.async = true;
-    var https = "https:" === document.location.protocol;
-    var prefix = (https ? "https://ssl" : "http://www");
-    ga.src = prefix + ".google-analytics.com/ga.js";
-    firstScript = document.getElementsByTagName("script")[0];
-    firstScript.parentNode.insertBefore(ga, firstScript);
-  };
-
-  GoogleAnalytics.trackPageview = function(url) {
-    if (!GoogleAnalytics.isLocalRequest()) {
-      if (url) {
-        window._gaq.push(["_trackPageview", url]);
-      } else {
-        window._gaq.push(["_trackPageview"]);
-      }
-      return window._gaq.push(["_trackPageLoadTime"]);
-    }
-  };
-
-  GoogleAnalytics.isLocalRequest = function() {
-    return GoogleAnalytics.documentDomainIncludes("local");
-  };
-
-  GoogleAnalytics.documentDomainIncludes = function(str) {
-    return document.domain.indexOf(str) !== -1;
-  };
-
-  GoogleAnalytics.getAnalyticsId = function() {
-    return $("[data-analytics-id]").data("analytics-id");
-  };
-
-  return GoogleAnalytics;
-
-})();
+// gtag intial setup
+window.dataLayer = window.dataLayer || [];
+function gtag(){dataLayer.push(arguments);}
+gtag('js', new Date());
 
 $(document).on("turbolinks:load", function(){
-  GoogleAnalytics.load();
-  if (GoogleAnalytics.analyticsId){
-    GoogleAnalytics.trackPageview();
+  // gtag config
+  const config = {}
+  // To turn off debug mode, exclude the parameter altogether (cannot just set to false)
+  // See https://support.google.com/analytics/answer/7201382?hl=en#zippy=%2Cgoogle-tag-websites
+  if (document.head.querySelector("meta[name=analytics_debug]").getAttribute('value') === "true") {
```

Comment:
It looked like for the other migrations, at least the ones I saw, we just put this script in the application template. Curious why in this case you've opted to retain the compiled analytics.js? By moving the script to the template you'd also avoid having to write/read the analytics debug setting from the DOM.